packages/flagging/src/openfeature/provider.ts

```diff
@@ -0,0 +1,61 @@
+/* eslint-disable local-rules/disallow-side-effects */
+import type { EvaluationContext, JsonValue, Logger, ResolutionDetails } from '@openfeature/core'
+import { StandardResolutionReasons } from '@openfeature/core'
+import type { Provider } from '@openfeature/web-sdk'
+
+export function createDatadogProvider(): Provider {
```

Comment:
Instead of a function, I think it's better export a class (e.g., `DatadogProvider` or `DatadogFeatureProvider`)

That seems simpler, and aligns better with OF documentation. e.g., example [here](https://openfeature.dev/specification/sections/flag-evaluation#requirement-1121):
> ```js
> // example provider mutator
> OpenFeature.setProvider(new MyProvider()); 
> ```

Reply:
I agree but this repository linter doesn't allow classes - I haven't dug into why yet but maybe we can make an exception for this package.